src/cpp/fastdds/topic/DDSSQLFilter/DDSFilterExpressionParserImpl/identifiers.hpp

```diff
@@ -48,32 +48,61 @@ struct identifier_processor
         return process_bound(bound_seq[0]);
     }
 
+    static xtypes::TypeIdentifier resolve_type(
+            const xtypes::TypeIdentifier& ti,
+            const position& pos)
+    {
+        if (xtypes::EK_COMPLETE != ti._d())
+        {
+            return ti;
+        }
+
+        std::shared_ptr<xtypes::TypeObject> type_object = std::make_shared<xtypes::TypeObject>();
+        if (RETCODE_OK == DomainParticipantFactory::get_instance()->type_object_registry().get_type_object(
+                    ti, *type_object) && xtypes::EK_COMPLETE == type_object->_d())
+        {
+            if (xtypes::TK_ALIAS == type_object->complete()._d())
+            {
+                const xtypes::TypeIdentifier& aliased_id =
+                        type_object->complete().alias_type().body().common().related_type();
+                return resolve_type(aliased_id, pos);
+            }
+            return ti;
+        }
+        throw parse_error("could not find type object definition", pos);
+    }
+
     static bool type_should_be_indexed(
             const xtypes::TypeIdentifier& ti,
             std::shared_ptr<xtypes::TypeIdentifier>& out_type,
-            size_t& max_size)
+            size_t& max_size,
+            const position& pos)
     {
         max_size = 0;
 
         switch (ti._d())
         {
             case xtypes::TI_PLAIN_ARRAY_SMALL:
-                out_type = std::make_shared<xtypes::TypeIdentifier>(*ti.array_sdefn().element_identifier());
+                out_type = std::make_shared<xtypes::TypeIdentifier>(
+                    resolve_type(*ti.array_sdefn().element_identifier(), pos));
                 max_size = process_bounds(ti.array_sdefn().array_bound_seq());
                 return true;
 
             case xtypes::TI_PLAIN_ARRAY_LARGE:
-                out_type = std::make_shared<xtypes::TypeIdentifier>(*ti.array_ldefn().element_identifier());
+                out_type = std::make_shared<xtypes::TypeIdentifier>(
+                    resolve_type(*ti.array_ldefn().element_identifier(), pos));
                 max_size = process_bounds(ti.array_ldefn().array_bound_seq());
                 return true;
 
             case xtypes::TI_PLAIN_SEQUENCE_SMALL:
-                out_type = std::make_shared<xtypes::TypeIdentifier>(*ti.seq_sdefn().element_identifier());
+                out_type = std::make_shared<xtypes::TypeIdentifier>(
+                    resolve_type(*ti.seq_sdefn().element_identifier(), pos));
                 max_size = process_bound(ti.seq_sdefn().bound());
                 return true;
 
             case xtypes::TI_PLAIN_SEQUENCE_LARGE:
-                out_type = std::make_shared<xtypes::TypeIdentifier>(*ti.seq_ldefn().element_identifier());
+                out_type = std::make_shared<xtypes::TypeIdentifier>(
+                    resolve_type(*ti.seq_ldefn().element_identifier(), pos));
```

Comment:
Might be a good idea to change `resolve_type` so it directly returns the shared_ptr